### PR TITLE
Fixed #25382 -- Removed obsolete references to DateQuerySet.

### PR DESCRIPTION
**Pull request by renatooliveira**

### CODE REVIEW

**Comment by timgraham** [comment]
There are some other mentions of DateQuerySet -- could you please remove them all?
```
docs/ref/class-based-views/generic-date-based.txt:      :meth:`DateQuerySet<django.db.models.query.QuerySet.dates>` object
docs/ref/class-based-views/generic-date-based.txt:      :meth:`DateQuerySet<django.db.models.query.QuerySet.dates>` object
docs/ref/class-based-views/generic-date-based.txt:      :meth:`DateQuerySet<django.db.models.query.QuerySet.dates>` object
docs/ref/models/querysets.txt:Returns a ``DateQuerySet`` — a ``QuerySet`` that evaluates to a list of
tests/aggregation_regress/tests.py:        # Regression for #10248 - Annotations work with DateQuerySets
tests/queries/tests.py:        # Pickling of DateQuerySets used to fail
tests/queries/tests.py:        # Raise proper error when a DateQuerySet gets passed a wrong type of
```

**Comment by renatooliveira** [comment]
yes, sure!

File: tests/queries/tests.py [hunk 1]
```diff
@@ -751,7 +751,7 @@ def test_ticket7791(self):
             3
         )
 
-        # Pickling of DateQuerySets used to fail
+        # Pickling of QuerySets used to fail
```
Comment (charettes) [inline tests/queries/tests.py]: Pickling of a QuerySet using dates()...

File: tests/queries/tests.py [hunk 2]
```diff
@@ -1302,7 +1302,7 @@ def test_ticket7107(self):
         self.assertQuerysetEqual(Valid.objects.all(), [])
 
     def test_ticket8683(self):
-        # Raise proper error when a DateQuerySet gets passed a wrong type of
+        # Raise proper error when a QuerySet gets passed a wrong type of
```
Comment (charettes) [inline tests/queries/tests.py]: ... when QuerySet.datetimes() gets passed ...

**Comment by renatooliveira** [comment]
@charettes fixed, thanks

**Comment by timgraham** [comment]
merged in e3720b990a33ae259da4b1f1f6069aa6bbc8c03d, thanks!

**Comment by renatooliveira** [comment]
Thanks for the feedback @timgraham and @charettes !

**Pull request closed without merging**